podpac/core/data/rasterio_source.py

```diff
@@ -35,12 +35,25 @@ class Rasterio(LoadFileMixin, BaseFileSource):
      crs : str, optional
         The coordinate reference system. Normally this will come directly from the file, but this allows users to
         specify the crs in case this information is missing from the file.
+    read_as_filename : bool, optional
+        Default is False. If True, the file will be read using rasterio.open(self.source) instead of being automatically
+        parsed to handle ftp, s3, in-memory files, etc. 
     """
 
     # dataset = tl.Instance(rasterio.DatasetReader).tag(readonly=True)
     band = tl.CInt(allow_none=True).tag(attr=True)
     crs = tl.Unicode(allow_none=True, default_value=None).tag(attr=True)
     driver = tl.Unicode(allow_none=True, default_value=None)
+    read_from_source = tl.Bool(False).tag(attr=True)
+
+    @cached_property
+    def dataset(self):
+        try:
```

Comment:
Can you make this a more explicit check? E.g. using something like `re.match(self.source, 'HDF4_EOS:EOS_GRID:.*:.*:.*)`

Reply:
It's a little tricky as gdal support multiple drivers, some that do not start with HDF4 (there's a netcdf flavor that I came across). But yes, I'll try to do something a bit more explicit.